web/src/WifiSelector.jsx

```diff
@@ -0,0 +1,276 @@
+  }, [client.network]);
+
+  useEffect(() => {
+    const loadNetworks = async () => {
+      const knownSsids = [];
+
+      return client.network.accessPoints()
+        .sort((a, b) => b.strength - a.strength)
+        .reduce((networks, ap) => {
+          // Do not include networks without SSID
+          if (!ap.ssid || ap.ssid === "") return networks;
+          // Do not include "duplicates"
+          if (knownSsids.includes(ap.ssid)) return networks;
+
+          const network = {
+            ...ap,
+            settings: connections.find(c => c.wireless?.ssid === ap.ssid),
+            connection: activeConnections.find(c => c.name === ap.ssid)
+          };
+
+          // Group networks
+          if (network.connection) {
+            networks.connected.push(network);
+          } else if (network.settings) {
+            networks.configured.push(network);
+          } else {
+            networks.others.push(network);
+          }
+
+          knownSsids.push(network.ssid);
+
+          return networks;
+        }, { connected: [], configured: [], others: [] });
+    };
+
+    loadNetworks().then((data) => {
+      setNetworks(data);
+      setActiveNetwork(networksFromValues(data).find(d => d.connection));
+    });
+  }, [client.network, connections, activeConnections]);
+
+  useEffect(() => {
+    return client.network.onNetworkEvent(({ type, payload }) => {
+      switch (type) {
+        case NetworkEventTypes.CONNECTION_ADDED: {
+          setConnections(conns => [...conns, payload]);
+          break;
+        }
+
+        case NetworkEventTypes.CONNECTION_UPDATED: {
+          setConnections(conns => {
+            const newConnections = conns.filter(c => c.id !== payload.id);
+            return [...newConnections, payload];
+          });
+          break;
+        }
+
+        case NetworkEventTypes.CONNECTION_REMOVED: {
+          setConnections(conns => conns.filter(c => c.path !== payload.path));
+          break;
+        }
+
+        case NetworkEventTypes.ACTIVE_CONNECTION_ADDED: {
+          setActiveConnections(conns => [...conns, payload]);
+          break;
+        }
+
+        case NetworkEventTypes.ACTIVE_CONNECTION_UPDATED: {
+          setActiveConnections(conns => {
+            const newConnections = conns.filter(c => c.id !== payload.id);
+            return [...newConnections, payload];
+          });
+          break;
+        }
+
+        case NetworkEventTypes.ACTIVE_CONNECTION_REMOVED: {
+          setActiveConnections(conns => conns.filter(c => c.id !== payload.id));
+          if (selectedNetwork?.settings?.id === payload.id) switchSelectedNetwork(null);
+          break;
+        }
+      }
+    });
+  });
+
+  const isStateChanging = (network) => {
+    const state = network.connection?.state;
+    return state === ConnectionState.ACTIVATING || state === ConnectionState.DEACTIVATING;
+  };
+
+  const renderFilteredNetworks = (networks) => {
+    return networks.map(n => {
```

Comment:
Please, use a meaningful name. For a small function, is perfectly fine, but this one has around 60 lines :-)